src/components/LoginForm/LoginForm.js

```diff
@@ -18,25 +23,44 @@ export class LoginForm extends Component {
   }
 
   handleLogin = (e) => {
-    const { addUser } = this.props;
     const { username, password } = this.state;
     const user = {
       username,
       password
     };
-
+    this.setState({ error: '', loading: true });
+    
     e.preventDefault();
+    this.signIn(user);
+  }
 
-    addUser(user);
+  signIn = async (user) => {
+    const { addUser } = this.props;
+
+    try {
+      const result = await requests.signIn(user);
+      const { user: validUser, token } = result.signIn;
+      if (token && validUser) {
+        localStorage.setItem('token', JSON.stringify(token));
+        addUser(validUser);
+        this.setState({ success: true, loading: false });
```

Comment:
Same goes for local error: we have a errored state in Redux, do we want to utilize that instead of local state?

Reply:
Same reasons here as stated from your previous question. I am also okay with it converting to connected dispatch but I think it's unnecessarily complex if it's just a temporary use case for the loading and error which leads to alot more  testing than necessary if we convert over to redux

Reply:
I think if we end up pulling async functionality to a thunk then we will want to dispatch error handling and loading handling to Redux, otherwise, if the async functionality is local we can handle those states locally as well. We might need to come to a consensus on this!